### PR TITLE
Get user data from Launch Configuration Group

### PR DESCRIPTION
**Pull request by gmendonca**
Yo!

This is an enhancement to allow tiopatinhas to get the user data from the Launch Configuration Group used by the Auto Scaling Group that tiopatinhas is monitoring. So, it will first check the class initialization, then the user data file passed by the configuration file and then it will check the Launch Configuration Group if none of those values are configured.

### CODE REVIEW

**Comment by gmendonca** [comment]
@barata @mwconceicao

File: tp/tp.py [hunk 1]
```diff
@@ -106,13 +108,23 @@ def __init__(self, side_group, weight_factor=1.0, debug=False,
         self.user_data = user_data
         user_data_file = self.conf.get("user_data_file", None)
         if not user_data and user_data_file:
+            self.logger.info("Trying to user data from file...")
```
Comment (barata) [inline tp/tp.py]: English problem :-)

File: tp/tp.py [hunk 2]
```diff
@@ -106,13 +108,23 @@ def __init__(self, side_group, weight_factor=1.0, debug=False,
...
             try:
                 with open(user_data_file) as f:
                     self.user_data = f.read()
             except IOError:
                 self.logger.warn("Could not read user data file: %s. Will launch instances without user data.",
                                  user_data_file)
 
+        if not user_data and not user_data_file and self.tapping_group.user_data:
+            self.logger.info("Trying to user data from launch configuration group...")
```
Comment (barata) [inline tp/tp.py]: English problem.

File: tp/tp.py [hunk 3]
```diff
@@ -106,13 +108,23 @@ def __init__(self, side_group, weight_factor=1.0, debug=False,
...
         if not user_data and user_data_file:
+            self.logger.info("Trying to user data from file...")
             try:
                 with open(user_data_file) as f:
                     self.user_data = f.read()
             except IOError:
                 self.logger.warn("Could not read user data file: %s. Will launch instances without user data.",
                                  user_data_file)
 
+        if not user_data and not user_data_file and self.tapping_group.user_data:
```
Comment (barata) [inline tp/tp.py]: Why not only checking for `user_data` variable. You should try to set it, if it fails (`not user_data`), try using a different strategy.

**Comment by gmendonca** [comment]
@barata

**Pull request merged**